### PR TITLE
Organizationally Unique Identifier for 25G/50G auto negotiation

### PR DESCRIPTION
**Pull request by AshokDaparthi**

### CODE REVIEW

**Comment by kcudnik** [comment]
Uint32 cant be empty use 0 zero

**Comment by lguohan** [comment]
build failures, can you investigate?

**Comment by kcudnik** [comment]
Hex numbers are not supported yet i will ads that when i will be back

File: inc/saiport.h [hunk 1]
```diff
@@ -531,6 +539,18 @@ typedef enum _sai_port_attr_t
...
 
     /**
+     * @brief Query/Configure Port's Advertised OUI code
+     *
+     * Organizationally Unique Identifier for 25G/50G auto negotiation.
+     * Default is 0x6A737D for 25G Ethernet Consortium
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0x6A737D
```
Comment (kcudnik) [inline inc/saiport.h]: i already added support for hex numbers, just need to be checked in #432

Comment (kcudnik) [inline inc/saiport.h]: this change is already checked in, if you will rebuild it should pass

Comment (AshokDaparthi) [inline inc/saiport.h]: how to restart build?

Comment (stcheng) [inline inc/saiport.h]: git ci --amend to update the commit and then git push --force to retrigger the build.

Comment (AshokDaparthi) [inline inc/saiport.h]: @kcudnik - 
i already added support for hex numbers, just need to be checked in #432 
This is pushed to master, my pull request to 0.9.6. Can you push #432 to 0.9.6 also.

Comment (kcudnik) [inline inc/saiport.h]: Added PR #461

**Comment by AshokDaparthi** [comment]
@guohan - Can you review this

**Comment by kcudnik** [comment]
Not this guohan :) @lguohan

**Comment by AshokDaparthi** [comment]
@lguohan   - Please review this

File: inc/saiport.h (hunk 1 above)
Comment (lguohan) [inline inc/saiport.h]: add validonly metadata when the port speed is 25 or 50G?

Comment (AshokDaparthi) [inline inc/saiport.h]: @lguohan - We have not defined any enum for speeds

Comment (lguohan) [inline inc/saiport.h]: speed is defined in unit of Mbps, we can define the validonly when speed=25000.

File: inc/saiport.h [hunk 2]
```diff
@@ -539,6 +539,7 @@ typedef enum _sai_port_attr_t
      * @type sai_uint32_t
      * @flags CREATE_AND_SET
      * @default 0x6A737D
+     * @validonly SAI_PORT_ATTR_SPEED == 25000Mbps, 50000Mbps
```
Comment (kcudnik) [inline inc/saiport.h]: actually validonly flag supports enums and bool's now no uint compare is supported yet 
you can remove this flag or comment it out

**Comment by lguohan** [comment]
retest this please

**Comment by AshokDaparthi** [comment]
@kcudnik - Could you please help on this, I added valid only values as 25000 and 50000. I think i need to modify some to pass the build.

**Comment by kcudnik** [comment]
Accepting those values is more compleated, for now please comments out Your condition And reverse lasy change i will ads this support when i will be back from vacation i next week

**Comment by AshokDaparthi** [comment]
@kcudnik - Can you help

**Comment by kcudnik** [comment]
Hey, sorry for the delay, i will add feature for this in this week

**Comment by kcudnik** [comment]
I added support for number conditions in #547. Please update your PR.

**Comment by kcudnik** [comment]
test this please

**Comment by kcudnik** [comment]
Ahh,  you are adding this to v9.6.0, i added support in origin/master

**Pull request closed without merging**